src/kvstore/CompactionFilter.h

```diff
@@ -80,18 +80,12 @@ class KVCompactionFilterFactory : public rocksdb::CompactionFilterFactory {
     auto now = time::WallClock::fastNowInSec();
     if (context.is_full_compaction || context.is_manual_compaction) {
       LOG(INFO) << "Do full/manual compaction!";
-      lastRunCustomFilterTimeSec_ = now;
-      return std::make_unique<KVCompactionFilter>(spaceId_, createKVFilter());
     } else {
-      if (FLAGS_custom_filter_interval_secs >= 0 &&
-          now - lastRunCustomFilterTimeSec_ > FLAGS_custom_filter_interval_secs) {
-        LOG(INFO) << "Do custom minor compaction!";
-        lastRunCustomFilterTimeSec_ = now;
-        return std::make_unique<KVCompactionFilter>(spaceId_, createKVFilter());
-      }
-      LOG(INFO) << "Do default minor compaction!";
-      return std::unique_ptr<rocksdb::CompactionFilter>(nullptr);
+      // No worry, by default flush will not go through the custom compaction filter.
```

Comment:
1. Maybe we should check `level` here. 
2. Besides, I think when level < 4, we could also do the custom compaction now and then, with low frequency.

Reply:
> 1. Maybe we should check `level` here.
> 2. Besides, I think when level < 4, we should also do the custom compaction now and then, with low frequency.

But it is not possible to get the level here.

Reply:
But in your code now, if the level < 4, filter will also be called for each key, which is not a good way.

Reply:
IMO, this is a tradeoff. Although this is not the perfect idea, it should be better to leave the expired data not-really handled. What do you think? If there is better idea, it is appreciated.